packages/api-reference/src/components/ApiReference.vue

```diff
@@ -1,36 +1,102 @@
 <script setup lang="ts">
 import { createHead, useSeoMeta } from 'unhead'
-import { ref } from 'vue'
+import { computed, toRef, watch } from 'vue'
+import { toast } from 'vue-sonner'
 
-import { type ReferenceProps } from '../types'
+import { useDarkModeState, useReactiveSpec, useSnippetTargets } from '../hooks'
+import { useToasts } from '../hooks/useToasts'
+import { useGlobalStore } from '../stores'
+import { type ReferenceConfiguration, type ReferenceProps } from '../types'
+import CustomToaster from './CustomToaster.vue'
 import Layouts from './Layouts/'
 
 const props = defineProps<ReferenceProps>()
 
+// If content changes it is emitted to the parent component to be handed back in as a spec string
 defineEmits<{
   (e: 'updateContent', value: string): void
 }>()
 
-const content = ref('')
+// Set defaults as needed on the provided configuration
+const configuration = computed<ReferenceConfiguration>(() => ({
+  spec: {
+    content: undefined,
+    url: undefined,
+    ...props.configuration?.spec,
+  },
+  proxy: undefined,
+  theme: 'default',
+  showSidebar: true,
+  isEditable: false,
+  ...props.configuration,
+}))
+
+// Configure Reference toasts to use vue-sonner
+const { initializeToasts } = useToasts()
+initializeToasts((message) => {
+  toast(message)
+})
 
 // Create the head tag if the configuration has meta data
-if (props.configuration?.metaData) {
+if (configuration.value?.metaData) {
   createHead()
-  useSeoMeta(props.configuration.metaData)
+  useSeoMeta(configuration.value.metaData)
 }
 
-function handleUpdateContent(value: string) {
-  content.value = value
-  props.configuration?.onSpecUpdate?.(value)
+// ---------------------------------------------------------------------------/
+// HANDLE MAPPING CONFIGURATION TO INTERNAL REFERENCE STATE
+
+/** Helper utility to map configuration props to the ApiReference internal state */
```

Comment:
should this be moved out into a helper function outside of a vue file @geoffgscott ?